File: jashing/src/main/java/com/github/avarabyeu/jashing/events/GraphEvent.java
package com.github.avarabyeu.jashing.events;

import com.github.avarabyeu.jashing.core.JashingEvent;

import java.util.Collection;

/**
 * @author Andrei Varabyeu
 */
public class GraphEvent extends JashingEvent {
    private Collection<Point> points;

    public GraphEvent(Collection<Point> points) {
        this.points = points;
    }

    public Collection<Point> getPoints() {
        return points;
    }

    public void setPoints(Collection<Point> points) {
        this.points = points;
    }

    public static class Point {
        private Integer x;
        private Integer y;

        public Point(Integer x, Integer y) {
            this.x = x;
            this.y = y;
        }

        public Integer getX() {
            return x;
        }

        public void setX(Integer x) {
            this.x = x;
        }

        public Integer getY() {
            return y;
        }

        public void setY(Integer y) {
            this.y = y;
        }
    }
}

File: jashing/src/main/kotlin/com/github/avarabyeu/jashing/events/GraphEvent.kt
package com.github.avarabyeu.jashing.events

import com.github.avarabyeu.jashing.core.JashingEvent

/**
 * @author Andrei Varabyeu
 */
class GraphEvent(val points: Collection<GraphEvent.Point>) : JashingEvent() {

    class Point(val x: Int, val y: Int)
}
